webapp/channels/src/components/analytics/doughnut_chart.tsx

```diff
@@ -16,30 +17,56 @@ type Props = {
 const DoughnutChart: React.FC<Props> = ({title, width, height, data}) => {
     const canvasRef = useRef<HTMLCanvasElement | null>(null);
     const chartRef = useRef<Chart<'doughnut'> | null>(null);
+    const prevData = useRef(data);
 
     useEffect(() => {
-        if (!data || !canvasRef.current) {
-            return;
-        }
+        const isEqual = data === prevData.current || deepEqual(prevData.current, data);
```

Comment:
Putting this inside the effect is a great idea. The only thing is that the `data === prevData.current` check should have already happened during dependency check, so it is not needed here, so we can simplify this to only the deepEqual check.

Reply:
Done!